src/hooks/useDragAndDrop.test.ts

```diff
@@ -0,0 +1,91 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+import { createAction } from "../helpers/test";
+import { canDrag } from "./useDragAndDrop";
+
+describe("useDragAndDrop", () => {
+  describe(canDrag.name, () => {
+    it("returns `null` for first step", () => {
+      expect(
+        canDrag(0, [
+          { actions: [createAction("action-1")] },
+          { actions: [createAction("action-2"), createAction("action-3")] },
+          { actions: [createAction("action-4")] },
+        ])
+      ).toBeNull();
+    });
+
+    it("returns `false` if preceding, current, and following steps have only one element", () => {
+      expect(
+        canDrag(1, [
+          { actions: [createAction("action-1")] },
+          { actions: [createAction("action-2")] },
+          { actions: [createAction("action-3")] },
```

Comment:
Instead of repeating `createAction` and `{ actions: { ... } }` in all these tests, what do you think about adding a  `createActions` that uses `createAction`:

```ts
export function createAction(
  name: string,
  overrides?: Partial<ActionInContext>
): ActionInContext {
  return {
    action: {
      name,
      signals: [],
    },
    frameUrl: "https://www.elastic.co",
    isMainFrame: true,
    pageAlias: "pageAlias",
    ...(overrides ?? {}),
  };
}

export function createActions(
  actionNames: [][]string,
): { actions: []ActionInContext } {
    return actionNames.map(names => {
      return { actions: names.map(name => createAction(name)) };
    })
}

// Then:
expect(canDrag(1, [["action-1"], ["action-2"], ["action-3"]]).toBe(false)
```

PS: We seem not to be using the `overrides` btw, but we can keep it there.

Reply:
Have a look at 9d32691772ea3e64362503649f1de0f09a60ef76 - I have actually added creator functions for `Step` and `Steps`, as there are valid cases for all three of these in the tests. The tests could all be re-written to use `createSteps`, but that would require quite a bit of work for the same output and doesn't seem worth it to me.